clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -190,18 +190,30 @@ class SourceMappingRegion {
 
   bool isBranch() const { return FalseCount.has_value(); }
 
+  bool isMCDCBranch() const {
+    const auto *BranchParams = std::get_if<mcdc::BranchParameters>(&MCDCParams);
+    assert(BranchParams == nullptr || BranchParams->ID >= 0);
```

Comment:
I think it would be nice if this assert was not necessary. Is it possible to enforce that BranchParams->ID is never when we have MC/DC parameters?

Reply:
I've removed the assertion. (I will commit other changes in BranchParams and DecisionParams later)